parsl/config.py

```diff
@@ -92,9 +91,11 @@ def __init__(self,
                  monitoring: Optional[MonitoringHub] = None,
                  usage_tracking: bool = False,
                  initialize_logging: bool = True) -> None:
+        self._executors: Sequence[ParslExecutor]
         if executors is None:
-            executors = [ThreadPoolExecutor()]
-        self.executors = executors
+            self._executors = [ThreadPoolExecutor()]
+        else:
+            self._executors = executors
```

Comment:
Other than the typing and privatization, I don't see the need for the logic change.  The only change of substance is the underscore prefix; mypy should figure out the typing from the function signature with:
```python
if executors is None:
    executors = (ThreadPoolExecutor(),)  # still in keeping with signature
self._executors = executors  # Now set, per signature; no explicit typing necessary
```